service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -781,32 +693,35 @@ private AddressInfoDto addressInfoDtoBuilder(Order order) {
     }
 
     private List<BagForUserDto> bagForUserDtosBuilder(Order order) {
-        Map<Integer, Integer> amountOfBags = new HashMap<>();
-        if (!MapUtils.isEmpty(order.getAmountOfBagsOrdered()) && MapUtils.isEmpty(order.getConfirmedQuantity())
-            && MapUtils.isEmpty(order.getExportedQuantity())) {
-            amountOfBags = order.getAmountOfBagsOrdered();
-        }
-        if (!MapUtils.isEmpty(order.getAmountOfBagsOrdered()) && !MapUtils.isEmpty(order.getConfirmedQuantity())
-            && MapUtils.isEmpty(order.getExportedQuantity())) {
-            amountOfBags = order.getConfirmedQuantity();
-        }
-        if (!MapUtils.isEmpty(order.getAmountOfBagsOrdered()) && !MapUtils.isEmpty(order.getConfirmedQuantity())
-            && !MapUtils.isEmpty(order.getExportedQuantity())) {
-            amountOfBags = order.getExportedQuantity();
-        }
-        List<Bag> bags = bagRepository.findBagByOrderId(order.getId());
-        List<BagForUserDto> bagForUserDtos = new ArrayList<>();
-        Map<Integer, Integer> finalAmountOfBags = amountOfBags;
-        bags.forEach(bag -> {
-            BagForUserDto bagDto = modelMapper.map(bag, BagForUserDto.class);
-            BagTranslation bagTranslation = bagTranslationRepository.findBagTranslationByBag(bag);
-            bagDto.setService(bagTranslation.getName());
-            bagDto.setServiceEng(bagTranslation.getNameEng());
-            bagDto.setCount(finalAmountOfBags.get(bag.getId()));
-            bagDto.setTotalPrice(finalAmountOfBags.get(bag.getId()) * bag.getFullPrice());
-            bagForUserDtos.add(bagDto);
-        });
-        return bagForUserDtos;
+        Map<Integer, Integer> actualBagAmounts = getActualBagAmountsForOrder(order);
+        List<Bag> bagsForOrder = bagRepository.findBagByOrderId(order.getId());
```

Comment:
Do we really need to get bags from the bag, why not use the bags from Order ??

Reply:
Yes, because in the Order we only have bag amounts, but not prices, capacities, translations, etc.